examples/metadata_processing/plot_add_dict_entries.py

```diff
@@ -1,46 +1,36 @@
 """
-=====================================================
-Add dictionary items in the standard DICOM dictionary
-=====================================================
-
-This examples illustrates how to add dictionary items to the 'standard' DICOM
-dictionary dynamically.
-
-.. warning::
-
-   Note that this is not a recommended solution in general but it is useful as
-   a demonstration or for in-house programs only, or to get around elements
-   missing from pydicom's DICOM dictionaries.
-
+=============================================================
+Add private dictionary items to the standard DICOM dictionary
+=============================================================
+
+This examples illustrates how to add private dictionary items to the DICOM
+dictionary dynamically. This allows to add unknown private tags to a new dataset,
+or reading private tags from an existing dataset not present in the pydicom
+private dictionary.
 """
 
-# authors : Darcy Mason
-#           Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# authors : Darcy Mason and pydicom contributors
 # license : MIT
 
 
-from pydicom.datadict import DicomDictionary, keyword_dict
+from pydicom.datadict import add_private_dict_entries
 from pydicom.dataset import Dataset
 
 print(__doc__)
 
-# Define items as (VR, VM, description, is_retired flag, keyword)
-#   Leave is_retired flag blank.
+# Define items as (VR, VM, description, is_retired flag)
+# Leave is_retired flag blank.
 new_dict_items = {
-    0x10011001: ("UL", "1", "Test One", "", "TestOne"),
-    0x10011002: ("OB", "1", "Test Two", "", "TestTwo"),
-    0x10011003: ("UI", "1", "Test Three", "", "TestThree"),
+    0x10011001: ("UL", "1", "Test One", ""),
+    0x10011002: ("OB", "1", "Test Two", ""),
+    0x10011003: ("UI", "1", "Test Three", ""),
 }
 
-# Update the dictionary itself
-DicomDictionary.update(new_dict_items)
-
-# Update the reverse mapping from name to tag
-new_names_dict = dict([(val[4], tag) for tag, val in new_dict_items.items()])
-keyword_dict.update(new_names_dict)
+# add the entries to the private dictionary, using the correct private creator string
+add_private_dict_entries(private_creator="ACME 3.1", new_entries_dict=new_dict_items)
 
 # Test that it is working
-ds = Dataset()  # or could get one from dcmread, etc
+ds = Dataset()
 
 ds.TestOne = 42
 ds.TestTwo = "12345"
```

Comment:
Should not be able to set using a private tag - in fact, this is just going to `ds.__dict__` rather than `ds._dict` which holds the DICOM data elements.  The old example updated to the regular dict (not private), so I guess it worked out.

Reply:
You are right, I completely missed this. Should have run it first myself... And generally I shouldn't make PRs while sleepy. :/

Reply:
While adapting this I noticed that there is no really convenient method to add a known private tag to a dataset, unless I missed something.
Doing it via `private_block` needs the VR as parameter, and it is also a bit cumbersome to first have to get the private block if you just want to set a single tag.

I'll probably leave this for the weekend, so I won't make the same error again...

Reply:
I may add a convenience function some times later in a separate PR, it it makes sense. For now, just fixed the example.